### PR TITLE
Migrate to Storybook 7

### PR DESCRIPTION
**Pull request by odinr**
This pull request includes several commits that refactor and update various components in the codebase. The changes include style refactoring, type declaration fixes, export statement updates, dependency updates, and documentation additions.

Cloes: [AB#49369](https://statoil-proview.visualstudio.com/3dadd756-da44-4bb3-8874-330932214dff/_workitems/edit/49369)

### CODE REVIEW

File: packages/person/package.json [hunk 1]
```diff
@@ -41,7 +41,7 @@
   },
   "dependencies": {
     "@equinor/fusion-react-utils": "^2.1.0",
-    "@equinor/fusion-wc-person": "^2.5.1"
+    "@equinor/fusion-wc-person": "^2.6.0"
```
Comment (asbjornhaland) [inline packages/person/package.json:44]: This is more updated in the upstream.

File: packages/person/src/PersonSelect.tsx [hunk 2]
```diff
@@ -24,6 +24,7 @@ type ElementAtts = PropsWithChildren<
       | 'selectedId'
       | 'value'
       | 'variant'
+      | 'selectedPerson'
```
Comment (asbjornhaland) [inline packages/person/src/PersonSelect.tsx:27]: I think this is already merged in main.

File: storybook/.storybook/main.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  excludedProps.includes(prop.name)
+);
+
+const repoRoot = path.resolve(__dirname, '../..');
+
+const config: StorybookConfig = {
+  stories: [
+    '../src/stories_v7/**/*.stories.tsx', 
+    '../src/stories_v7/**/*.mdx',
+    // '../../packages/**/*.stories.tsx',
```
Comment (asbjornhaland) [inline storybook/.storybook/main.ts:39]: Please delete this line.

File: storybook/.storybook/main.ts [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    //   options: {
+    //     configureJSX: true,
+    //     babelOptions: {},
+    //     sourceLoaderOptions: null,
+    //   }
+    // }, 
+    // {
+    //   name: "@storybook/addon-docs",
+    //   options: { transcludeMarkdown: true },
+    // },
```
Comment (asbjornhaland) [inline storybook/.storybook/main.ts:98]: Is this commented code necessary?

File: storybook/.storybook/preview.tsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+// };
+
+// export const decorators = [
+//   withPropsTable,
+//   (Story) => (
+//     <ThemeProvider theme={theme}>
+//       <Story />
+//      </ThemeProvider>
+//   ),
+// ];
```
Comment (asbjornhaland) [inline storybook/.storybook/preview.tsx:90]: Is this commented code necessary?

File: storybook/src/components/PackageInfo.tsx [hunk 6]
```diff
@@ -52,16 +53,22 @@ export const renderDependencies = (deps: Record<string, string>) => {
...
+export const PackageInfo = ({ pkg }: { readonly pkg: Package }): React.ReactElement => {
   return (
     <div>
-      <Title>{pkg.name}</Title>
+      <h1>{pkg.name}</h1>
       <a title="Published on npm" href={`https://www.npmjs.com/package/${pkg.name}`}>
-        <img src={`https://img.shields.io/npm/v/${pkg.name}.svg`} />
+        <img alt={pkg.name} src={`https://img.shields.io/npm/v/${pkg.name}.svg`} />
       </a>
-      <Subtitle>{pkg.description}</Subtitle>
```
Comment (asbjornhaland) [inline storybook/src/components/PackageInfo.tsx:62]: Shouldn't package description be visible anymore?

File: storybook/package.json [hunk 7]
```diff
@@ -35,33 +36,45 @@
...
-    "@storybook/addon-links": "^6.5.16",
-    "@storybook/addon-storysource": "^6.5.16",
-    "@storybook/builder-webpack5": "^6.5.16",
-    "@storybook/manager-webpack5": "^6.5.16",
-    "@storybook/react": "^6.5.16",
-    "babel-loader": "^9.1.2",
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2",
+    "@equinor/fusion-wc-person": "^2.6.0",
+    "@equinor/fusion-wc-theme": "^1.1.0",
```
Comment (asbjornhaland) [inline storybook/package.json:40]: Are these necessary?

File: storybook/src/stories_v7/context-selector/component.mdx.tmp
Comment (asbjornhaland) [inline storybook/src/stories_v7/context-selector/component.mdx.tmp:1]: Why is this a `.tmp` file?

File: storybook/src/stories_v7/context-selector/context-selector.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+// export const ContextHeader: Story = ({ children, ...props }: React.PropsWithChildren<ContextSearchProps>) => {
+//   return (
+//     <ContextProvider resolver={_exampleResolver}>
+//       <ContextSearch {...props}>{children}</ContextSearch>
+//     </ContextProvider>
+//   );
+// };
+// ContextHeader.args = {
+
+// };
```
Comment (asbjornhaland) [inline storybook/src/stories_v7/context-selector/context-selector.stories.tsx:50]: Please remove dead code.

File: storybook/src/stories_v7/textarea/textarea.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+    disabled: true,
+  },
+};
+
+// export const charCounter: Story = {
+//   args: {
+//     label: 'Character counter',
+//     charCounter: true,
+//   },
+// };
```
Comment (asbjornhaland) [inline storybook/src/stories_v7/textarea/textarea.stories.tsx:46]: Please uncomment or remove.

**Review by asbjornhaland — CHANGES_REQUESTED** [review]
Only inspected the code. Will test the storybook locally tomorrow.

**Comment by asbjornhaland** [comment]
Also please rebase on `main`.

File: storybook/src/stories_v7/textarea/textarea.stories.tsx (hunk 9 above)
Comment (odinr) [reply storybook/src/stories_v7/textarea/textarea.stories.tsx:46]: nope, that story should work, so leave trace for future fix, no time atm

File: packages/person/package.json (hunk 1 above)
Comment (odinr) [reply packages/person/package.json]: fixed with last rebase

File: packages/person/src/PersonSelect.tsx (hunk 2 above)
Comment (odinr) [reply packages/person/src/PersonSelect.tsx]: order fixed

File: storybook/src/stories_v7/context-selector/context-selector.stories.tsx
[the remark this answers is not in the document]
Comment (odinr) [reply storybook/src/stories_v7/context-selector/context-selector.stories.tsx:1]: works hith search for "joh", was like this before upgrade, wnot fix, can you create an issue for making faker for context search

**Review by asbjornhaland — APPROVED** [review]

**Pull request merged**